filters/include/pcl/filters/experimental/passthrough.h

```diff
@@ -0,0 +1,215 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2021-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/filters/experimental/functor_filter.h>
+
+#include <limits> // for numeric_limits
+
+namespace pcl {
+namespace experimental {
+
+template <typename PointT>
+struct PassThroughFunctor {
+  PassThroughFunctor() = default;
+  PassThroughFunctor(const uindex_t* field_offset,
+                     const float* filter_limit_min,
+                     const float* filter_limit_max,
+                     const bool* has_filter_field,
+                     const bool* negative)
+  : field_offset_(field_offset)
+  , filter_limit_min_(filter_limit_min)
+  , filter_limit_max_(filter_limit_max)
+  , has_filter_field_(has_filter_field)
+  , negative_(negative)
+  {}
+
+  bool
+  operator()(const PointCloud<PointT>& cloud, index_t idx) const
+  {
+    if (!*has_filter_field_)
+      return !*negative_;
+
+    float field_value;
+    const std::uint8_t* pt_data = reinterpret_cast<const std::uint8_t*>(&cloud.at(idx));
+    memcpy(&field_value, pt_data + *field_offset_, sizeof(float));
+    if (!std::isfinite(field_value))
+      return *negative_;
```

Comment:
Why do you not `return false` here?

Reply:
returning `negative_` can make sure `nan` points are always removed in `FunctorFilter`
(L99 will be always `false` with any value of `negative_`)

https://github.com/PointCloudLibrary/pcl/blob/fad821ef31eb92719c1005812666136f9664d365/filters/include/pcl/filters/experimental/functor_filter.h#L97-L106

in the original `PassThrough` filtering `nan` points is not affected by `negative_`

https://github.com/PointCloudLibrary/pcl/blob/fad821ef31eb92719c1005812666136f9664d365/filters/include/pcl/filters/impl/passthrough.hpp#L55-L71